Directory.Build.targets

```diff
@@ -86,10 +86,21 @@
   </ItemGroup>
 
   <!-- Include docs xmls in the final packages. -->
-  <Target Name="AddDocumentationFileForPackaging" AfterTargets="DocumentationProjectOutputGroup">
+  <Target Name="AddDocumentationFileForPackaging"
+          AfterTargets="DocumentationProjectOutputGroup"
+          Condition="'$(GenerateDocumentationFile)' == 'false'">
     <ItemGroup>
       <DocumentationProjectOutputGroupOutput Include="$(XmlDocumentationFilePath)" />
     </ItemGroup>
   </Target>
+  
+    <!-- Allow P2Ps to retrieve the DocFileItem path. -->
+  <Target Name="AnnotateTargetPathWithTargetPlatformMonikerWithDocFileItem"
+          DependsOnTargets="GetTargetPathWithTargetPlatformMoniker"
+          BeforeTargets="GetTargetPath">
+    <ItemGroup>
+      <TargetPathWithTargetPlatformMoniker DocFileItem="@(DocFileItem)" />
+    </ItemGroup>
+  </Target>
```

Comment:
Why do you need this target?

Reply:
So after building, I looked at all the places where an xml was expected. Before my change, I saw that the folder: `artifacts/bin/System.Memory/<Config>/<TargetFramework>/` had System.Memory.dll/xml and System.Buffers.dll/xml files. After my change (without the above target) I noticed the System.Buffers.xml was not being copied anymore. So I investigated and realized this target was taking care of that in runtime. I built again with this target and now the System.Buffers.xml file is showing up in that folder again.

Reply:
Note: I confirmed that this target didn't affect the nupkg output. The xmls can still be found in the `include/` folders.

Reply:
Thanks for following-up. I'm not sure I understand what you mean. Do you agree that the target isn't necessary? Just from looking at it, it shouldn't have any impact on the build in this repo as no one uses the `DocFileItem` metadata on `TargetPathWithTargetPlatformMoniker` items. This is only needed in runtime for this target: https://github.com/dotnet/runtime/blob/b8816bcae29bd1aae8ed421075142f1ab178f604/src/libraries/sfx.proj#L63

Reply:
Ok after showing you on video, I was wrong and the target was not what was causing the System.Buffers.xml to not get copied to the System.Memory folder. I don't know exactly what caused it yesterday. I pushed a commit to remove this target.